src/table/table.component.ts

```diff
@@ -843,4 +916,86 @@ export class Table {
 		event.target.parentElement.parentElement.parentElement.parentElement.children[1].scrollTop = 0;
 		this.model.isEnd = false;
 	}
+
+	handleTabIndex() {
+		setTimeout(() => {
+			const focusElementList = getFocusElementList(this.elementRef.nativeElement, tabbableSelectorIgnoreTabIndex);
+			if (focusElementList.length > 0) {
+				focusElementList.forEach(tabbable => {
+					tabbable.tabIndex = -1;
+				});
+			}
+			Array.from<HTMLElement>(this.elementRef.nativeElement.querySelectorAll("td, th")).forEach(cell => setTabIndex(cell, -1));
+
+			const rows = this.elementRef.nativeElement.firstElementChild.rows;
+			if (Array.from(rows[0].querySelectorAll("th")).some(th => getFocusElementList(th, tabbableSelectorIgnoreTabIndex).length > 0)) {
+				setTabIndex(rows[0].querySelector("th"), 0);
+			} else {
+				setTabIndex(rows[1].querySelector("td"), 0);
+			}
+		});
+	}
+
+	getTotalColumns() {
+		if (this.model.hasExpandableRows() && this.showSelectionColumn) {
+			return this.model.header.length + 2;
+		} else if (this.model.hasExpandableRows() || this.showSelectionColumn) {
+			return this.model.header.length + 1;
+		} else {
+			return this.model.header.length;
+		}
+	}
+
+	// This is handled in the component so that the columnIndex is saved as you go into expandable rows
+	handleInteractions(event: KeyboardEvent) {
```

Comment:
Directive can know (based on classList) if it's extended row. `columnIndex` can use double binding to update values. This won't be necessary.